### PR TITLE
Updating kubernetes version

### PR DESCRIPTION
**Pull request by davidjumani**
Fixes https://github.com/apache/cloudstack-kubernetes-provider/issues/28

### CODE REVIEW

**Comment by davidjumani** [comment]
@onitake @joschi36 Can you review ?
Can use the image over at `davidjumani/cloudstack-kubernetes-provider:k-v1.18.18`
Thanks

**Review by rhtyd — APPROVED** [review]
LGTM need build confirmation.
@davidjumani can you check/add/update container/project building with travis? https://travis-ci.com/github/apache/cloudstack-kubernetes-provider

**Comment by davidjumani** [comment]
@rhtyd Added docker builds to travis. Unit tests are needed which is tracked in https://github.com/apache/cloudstack-kubernetes-provider/issues/23

File: Makefile [hunk 1]
```diff
@@ -34,7 +34,7 @@ CMD_SRC=\
 
 .PHONY: all clean docker
 
-all: cloudstack-ccm
+all: clean test cloudstack-ccm
```
Comment (onitake) [inline Makefile:37]: Please don't put that here.
If you think a "all" build should delete build artifacts and run the unit tests, open a separate PR.

File: Dockerfile [hunk 2]
```diff
@@ -18,7 +18,7 @@
 FROM golang:1.13 as builder
 COPY . /go/src/github.com/apache/cloudstack-kubernetes-provider
 WORKDIR /go/src/github.com/apache/cloudstack-kubernetes-provider
-RUN make clean && CGO_ENABLED=0 GOOS=linux make
+RUN CGO_ENABLED=0 GOOS=linux make
```
Comment (onitake) [inline Dockerfile:21]: See comment below

File: .travis.yml [hunk 3]
```diff
@@ -4,4 +4,7 @@ go:
 - 1.13
 - master
 
-script: make test
+services:
+  - docker
+
+script: make docker
```
Comment (onitake) [inline .travis.yml:10]: See below.

**Review by onitake — CHANGES_REQUESTED** [review]
I don't think we should change build behavior with this PR.
Updating k8s dependencies LGTM, but the rest needs further elaboration.
Please open a new PR for that.

**Comment by joschi36** [comment]
Hey @davidjumani

> Can use the image over at davidjumani/cloudstack-kubernetes-provider:k-v1.18.18

Is it possible that it's not build as a non-root container, as I can't start it on my cluster.

`
Error: failed to start container "cloud-controller-manager": Error response from daemon: OCI runtime create failed: container_linux.go:346: starting container process caused "exec: \"/root/cloudstack-ccm\": stat /root/cloudstack-ccm: permission denied": unknown
`

File: .travis.yml (hunk 3 above)
Comment (onitake) [reply .travis.yml:10]: The correct wait to run both tests and a release build in Travis is to use build stages: https://docs.travis-ci.com/user/build-stages/

**Comment by davidjumani** [comment]
@joschi36 I've rebuilt it, can you pull it again and try ?
@onitake Reverted the change and will open a separate PR

**Comment by onitake** [comment]
Thanks, @davidjumani .
LGTM from a formal point of view, but please wait for test results from @joschi36 .

**Review by onitake — APPROVED** [review]

**Comment by rhtyd** [comment]
Thanks @onitake, we'll wait to hear from @joschi36 on tests

**Comment by joschi36** [comment]
@davidjumani When running in my setup I had to change RBAC rules. 

Got this error:

```I0503 12:44:20.142166       1 leaderelection.go:247] failed to acquire lease kube-system/cloud-controller-manager```
```E0503 12:44:23.753152       1 leaderelection.go:324] error retrieving resource lock kube-system/cloud-controller-manager: leases.coordination.k8s.io "cloud-controller-manager" is forbidden: User "system:serviceaccount:kube-system:cloud-controller-manager" cannot get resource "leases" in API group "coordination.k8s.io" in the namespace "kube-system"```


Maybe we need to update them aswell in the file [deployment.yaml](https://github.com/apache/cloudstack-kubernetes-provider/blob/master/deployment.yaml). Or even better, create a new Role and RoleBinding limiting this change to only the Namespace.
```diff
-- apiGroups:
-  - ""
-  resources:
-  - endpoints
-  verbs:
-  - create
-  - get
-  - list
-  - watch
-  - update
+- apiGroups:
+  - coordination.k8s.io
+  resources:
+  - leases
+  verbs:
+  - get
+  - create
+  - update
```

Sorry for the late review, I'm very busy at the moment.

**Comment by onitake** [comment]
@joschi36 I think this issue is not tied to the minor version bump, but was introduced with k8s 1.17.
I know that you are busy, but could you test the proposed role change and open a separate PR?
Or, I could do it if you tell me (via PM) which cluster I can test this on.

**Review by joschi36 — APPROVED** [review]
@onitake After I changed the rbac it worked. :+1:

**Pull request merged**